### PR TITLE
remove compiler method guice binding

### PR DESCRIPTION
**Pull request by parthshyara**
- Removing guice binding to the synthetic and bridged methods generated by the compiler to prevent duplicate profiling.
- Refactored log printing.

### CODE REVIEW

File: modules/gojira-core/src/main/java/com/flipkart/gojira/core/BindingsModule.java [hunk 1]
```diff
@@ -38,8 +40,15 @@ protected void configure() {
...
-          Matchers.annotatedWith(ProfileOrTest.class),
+          new NoCompilerMethodMatcher().and(Matchers.annotatedWith(ProfileOrTest.class)),
           profileOrTestMethodInterceptor);
     }
   }
+
+  private static class NoCompilerMethodMatcher extends AbstractMatcher<Method> {
+    @Override
+    public boolean matches(Method method) {
+      return !method.isSynthetic() && !method.isBridge();
```
Comment (balalakshmifk) [inline modules/gojira-core/src/main/java/com/flipkart/gojira/core/BindingsModule.java:51]: does this include lambdas as well?

Comment (parthshyara) [reply modules/gojira-core/src/main/java/com/flipkart/gojira/core/BindingsModule.java:51]: No, it don't think it does. 
Reference: https://medium.com/datadriveninvestor/java-synthetic-methods-what-are-these-52f77efb347a.

**Review by balalakshmifk — APPROVED** [review]
LGTM. No changes required from AspectJ side. Tested bridge methods with aspectJ interception and it invokes only once.

**Review by arunachalams — APPROVED** [review]
lgtm.

**Pull request merged**